tools/tplg_parser/tokens.c

```diff
@@ -49,7 +49,7 @@ int get_token_uint32_t(void *elem, void *object, uint32_t offset,
 		       uint32_t size)
 {
 	struct snd_soc_tplg_vendor_value_elem *velem = elem;
-	uint32_t *val = object + offset;
+	uint32_t *val = (uint32_t *)((uint8_t *)object + offset);
```

Comment:
```suggestion
	uint32_t *val = (uint32_t *)object + offset/4;
```

Reply:
these aren't quite equivalent of course - with the former if `offset` isn't a multiple of 4, we get unaligned access

Reply:
I'll do a separate PR for this since it's a real bug fix and change the two other correct functions to similar pointer arithmetic.